daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/layers/api/ChecksumLayer.java

```diff
@@ -23,29 +23,29 @@
 /**
  * A checksum layer computes some sort of integer value from a region of the data stream. The term checksum is
  * used generically here to subsume all sorts of CRCs, check digits, data hash, and digest calculations.
- *
+ * <p/>
  * This base is suitable only for checksums computed over small sections of data, not large data streams or whole files.
- *
+ * <p/>
  * The entire region of data the checksum is being computed over, will be pulled into a byte buffer in memory.
- *
+ * <p/>
  * The resulting checksum is the return value of the compute method.
- *
+ * <p/>
  * This is delivered into a DFDL variable for use by the DFDL schema. This variable can have any name
  * such as 'crc', 'digest', or 'dataHash'.
- *
+ * <p/>
  * The derived implementation class must also define a getter method based on the name of the DFDL variable which
  * will be assigned with the checksum value. For example if the checksum is actually a specific digest/hash calculation
  * and the DFDL variable is named 'digest', then this getter must be defined:
- *
- *     Int getDFDLResultVariable_digest() { return getChecksum() }
- *
+ * <p/>
+ *     Int getLayerVariableResult_digest() { return getChecksum() }
+ * <p/>
  * This will be called automatically to retrieve the integer value that was returned from the `compute` method, and
  * the DFDL variable 'digest' will be assigned that value.
  */
 public abstract class ChecksumLayer extends ChecksumLayerBase {
```

Comment:
Do we need a comment that implementations must call `setLength` prior to wrap being called?

Reply:
fixed.